File: backend/src/dtos/group.ts
import { Group } from '@prisma/client';

export type GroupDto = Pick<Group, 'id' | 'name' | 'wordCount'>;

export type CreateGroupDto = Pick<Group, 'name' | 'userId'>;

export type UpdateGroupDto = Pick<Group, 'name' | 'id'>;

File: shared/contracts/dto/group.go
package dto

type Group struct {
	ID        string `json:"id"`
	Name      string `json:"name"`
	WordCount int    `json:"wordCount"`
}

type CreateGroup struct {
	Name string `json:"name"`
}

type UpdateGroup struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}
